js/modules/k6/experimental/fs/registry.go

```diff
@@ -0,0 +1,75 @@
+package fs
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+// registry is a registry of opened files.
+type registry struct {
```

Comment:
I'm not opposed to the `Cache` name indeed. I'll look into https://github.com/grafana/k6/pull/3165#issuecomment-1686420904, and if I end up not getting rid of it, I'll rename it accordingly 👍🏻

Reply:
>and if I end up not getting rid of it

If you mean dropping the added Registry concept then I think we should not do it. It would complicate the migration from `afero`. The added Registry is an abstraction from `afero` that should simplify the work for dropping it.
We said `afero` debt wouldn't be part of this work so I guess having one more copy is something we can have until we decide to drop completely `afero`. Except, if we can not use the afero layer only for this specific module, but it sounds unrealistic to me.

I hope I got right your sentence, otherwise, feel free to correct me.

Reply:
My reply was too short and needed to be more specific about the context.

I'm unkeen to remove the concept of a file "registry" (whatever we name it) because we have somewhat of an agreement that we want to reduce our dependency on Afero. 

Even though we currently [have the file content cached twice instead of once](https://github.com/grafana/k6/pull/3165#issuecomment-1686420904), once by ourselves, and a second time by the underlying afero fs `CacheOnRead`, it's still better than VU times, and it affords us to limit our dependency on afero itself, and keeps the path towards reducing our use of it easier.

I would rename `registry` as `Cache` for now 👍🏻